src/main/webapp/app/exercises/text/manage/text-exercise/text-exercise-detail.component.ts

```diff
@@ -106,4 +111,60 @@ export class TextExerciseDetailComponent implements OnInit, OnDestroy {
     registerChangeInTextExercises() {
         this.eventSubscriber = this.eventManager.subscribe('textExerciseListModification', () => this.load(this.textExercise.id));
     }
+
+    checkPlagiarismJson() {
+        this.checkPlagiarism((data) => {
+            const json = JSON.stringify(data);
+            const blob = new Blob([json], { type: 'application/json' });
+            downloadFile(blob, `check-plagiarism-${this.textExercise.id}.json`);
+        });
+    }
+
+    checkPlagiarismCsv() {
+        this.checkPlagiarism((data) => {
+            if (data.length > 0) {
+                const csvExporter = new ExportToCsv({
+                    fieldSeparator: ';',
+                    quoteStrings: '"',
+                    decimalSeparator: 'locale',
+                    showLabels: true,
+                    title: `Plagiarism Check for Exercise ${this.textExercise.id}: ${this.textExercise.title}`,
+                    filename: `check-plagiarism-${this.textExercise.id}`,
+                    useTextFile: false,
+                    useBom: true,
+                    headers: ['Student A', 'Submission A', 'Student B', 'Submission B', ...Object.keys(data[0].distanceMetrics)],
+                });
+
+                const fullname = (student: User): string => `${student.firstName || ''} ${student.lastName || ''}`;
```

Comment:
```suggestion
                const fullname = (student: User): string => `${student.firstName || ''}${' ' + student.lastName || ''}`;
```

Reply:
Is this solving trailing spaces? Otherwise, we could also call `.trim()`.

Reply:
I think calling .trim() would work even better.